src/beancount.pest

```diff
@@ -38,9 +38,9 @@ valid_non_letter_commodity_char = { QUOTATION_MARK |  "_" | HYPHEN | "." }
 commodity = @{ ASCII_ALPHA_UPPER ~ (ASCII_ALPHA_UPPER | valid_non_letter_commodity_char){0, 22} ~ (ASCII_ALPHA_UPPER | ASCII_DIGIT)? }
 
 //// Account primitives
-account_type = { "Assets" | "Liabilities" | "Income" | "Expenses" | "Equity" }
-account_name_piece = { ":" ~ (ASCII_ALPHA_UPPER | ASCII_DIGIT) ~ (ASCII_ALPHA_LOWER | ASCII_ALPHA_UPPER | ASCII_DIGIT | "-")* }
-account = @{ account_type ~ account_name_piece* }
+account_type = { UPPERCASE_LETTER ~ (LETTER | DECIMAL_NUMBER | "-")* }
+account_name_piece = { ":" ~ (UPPERCASE_LETTER | DECIMAL_NUMBER) ~ (LETTER | DECIMAL_NUMBER | "-")* }
+account = @{ account_type ~ account_name_piece+ }
```

Comment:
Is an account required to have at least one piece?  If so, TIL.

Reply:
Yep, see https://bitbucket.org/blais/beancount/src/487cb9b0248427ac6882889b6b8bcc1665ce393e/beancount/parser/lexer.l?at=default&fileviewer=file-view-default#lexer.l-279 🙂